### PR TITLE
Refactor and add JUnit for remove note

### PR DESCRIPTION
**Pull request by n3wsoldier**
1) Refactor BookmarkParser and BookmarkCommand (SLAP)
2) JUnit test for remove note valid and invalid

### CODE REVIEW

File: src/test/java/anichan/command/BookmarkCommandTest.java [hunk 1]
```diff
@@ -36,14 +36,20 @@
...
     private static final String VALID_SINGLE_INPUT_TEST = "1";
     private static final String VALID_SINGLE_INPUT_TEST2 = "2";
     private static final String VALID_SINGLE_INPUT_TEST3 = "3";
+    private static final String VALID_REMOVE_NOTE_TEST = "1 -r 2";
     private static final String INVALID_ADD_ANIME_INDEX_TEST = "-a 600";
     private static final String INVALID_DELETE_BOOKMARK_INDEX_TEST = "-d 5";
     private static final String INVALID_ADD_DUPLICATE_ANIME_INDEX_TEST = "-a 1";
     private static final String INVALID_EDIT_BOOKMARK_INDEX_TEST = "5 -e 5";
     private static final String INVALID_INFO_BOOKMARK_INDEX_TEST = "5";
+    private static final String INVALID_NOTE_BOOKMARK_INDEX_TEST = "5 -n test";
```
Comment (ChanJianHao) [inline src/test/java/anichan/command/BookmarkCommandTest.java:50]: Great work refactoring them1

File: src/test/java/anichan/command/BookmarkCommandTest.java [hunk 2]
```diff
@@ -193,22 +233,38 @@ void execute_validInfo_Successful() throws AniException {
...
+        BookmarkParser testParse = new BookmarkParser();
+        BookmarkCommand testBookmarkCommand = testParse.parse(VALID_REMOVE_NOTE_TEST);
+        String result = testBookmarkCommand.execute(animeData, storageManager, user);
+        assertEquals(OUTPUT_REMOVE_NOTE_TEST, result);
+        System.out.println(result);
+
+        BookmarkParser testParse2 = new BookmarkParser();
+        BookmarkCommand testBookmarkCommand2 = testParse2.parse(VALID_SINGLE_INPUT_TEST);
+        String result2 = testBookmarkCommand2.execute(animeData, storageManager, user);
+        assertEquals(OUTPUT_SINGLE_INPUT_TEST_AFTER_REMOVE, result2);
```
Comment (ChanJianHao) [inline src/test/java/anichan/command/BookmarkCommandTest.java:261]: Good tests!

File: text-ui-test/EXPECTED.TXT [hunk 3]
```diff
@@ -88,13 +88,13 @@ COOKIES N CREME CH33SECAKE (Default) #> Saving 3. Trigun to bookmark.
...
 COOKIES N CREME CH33SECAKE (Default) #> Saving 430. To Heart 2 to bookmark.
 
-COOKIES N CREME CH33SECAKE (Default) #> Editing Cowboy Bebop to have 5 episode
+COOKIES N CREME CH33SECAKE (Default) #> Editing Cowboy Bebop to have 5 episode(s).
 
-COOKIES N CREME CH33SECAKE (Default) #> Editing InuYasha the Movie 2: The Castle Beyond the Looking Glass to have 4 episode
+COOKIES N CREME CH33SECAKE (Default) #> Editing InuYasha the Movie 2: The Castle Beyond the Looking Glass to have 4 episode(s).
 
-COOKIES N CREME CH33SECAKE (Default) #> Editing Cowboy Bebop: The Movie - Knockin' on Heaven's Door to have 6 episode
+COOKIES N CREME CH33SECAKE (Default) #> Editing Cowboy Bebop: The Movie - Knockin' on Heaven's Door to have 6 episode(s).
```
Comment (ChanJianHao) [inline text-ui-test/EXPECTED.TXT:95]: Great addition!

**Review by ChanJianHao — APPROVED** [review]
lgtm!

**Review by EyoWeiChin — APPROVED** [review]
LGTM!

**Pull request merged**